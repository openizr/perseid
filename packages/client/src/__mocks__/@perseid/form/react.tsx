/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/**
 * `@perseid/form/react` mock.
 */

export default function Form(props: unknown): JSX.Element {
  return (
    <div id="react-form">
      <span>{JSON.stringify(props)}</span>
    </div>
  );
}
