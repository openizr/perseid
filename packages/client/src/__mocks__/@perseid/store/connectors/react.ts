/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/**
 * `@perseid/store/connectors/react` mock.
 */

export default vi.fn(() => vi.fn((module: string) => {
  if (module === 'router') {
    return (process.env.NOT_FOUND === 'true')
      ? { route: null }
      : { route: process.env.PAGE };
  }
  return (process.env.ERROR === 'true')
    ? new Error('test')
    : null;
}));
