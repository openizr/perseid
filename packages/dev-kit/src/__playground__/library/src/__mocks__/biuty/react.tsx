import React from 'react';

export function UIP(): JSX.Element {
  return <p>TEST</p>;
}

export function UITitle(): JSX.Element {
  return <p>TEST</p>;
}
