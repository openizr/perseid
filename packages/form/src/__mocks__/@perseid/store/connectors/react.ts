/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/**
 * @perseid/store/connectors/react mock.
 */
export default function useStore(): unknown {
  return vi.fn(() => ({
    steps: [{ id: 'start' }, { id: 'end' }],
    loading: process.env.LOADING === 'true',
  }));
}
