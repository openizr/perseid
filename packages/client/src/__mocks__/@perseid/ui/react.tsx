/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';

/**
 * `@perseid/ui/react` mock.
 */

interface ComponentProps {
  [key: string]: unknown;
  onClick?: (event: Event) => void;
  onBlur?: (value: string) => void;
  onFocus?: (value: string) => void;
  onChange?: (newValue: string) => void;
  onKeyDown?: (value: string, event: KeyboardEvent) => void;
  columns: { component: React.ReactNode; path: string; }[];
}

function Component(type: string): (props: ComponentProps) => JSX.Element {
  return function RealComponent({
    onBlur,
    onClick,
    onFocus,
    onChange,
    onKeyDown,
    ...props
  }) {
    React.useEffect(() => {
      // Covers `onChange` handler.
      if (onChange !== undefined) {
        onChange((props.value === '2023-01-01T00:00:00.000Z')
          ? '2023-01-02T00:00:00.000Z'
          : process.env.VALUE ?? 'test');
      }
      // Covers `onBlur` handler.
      if (onBlur !== undefined) {
        onBlur('test');
      }
      // Covers `onKeyDown` handler.
      if (onKeyDown !== undefined) {
        onKeyDown('test', { key: 'ArrowDown' } as KeyboardEvent);
      }
      // Covers `onFocus` handler.
      if (onFocus !== undefined) {
        onFocus('test');
      }
      // Covers `onClick` handler.
      if (onClick !== undefined) {
        onClick({ preventDefault: vi.fn() } as unknown as Event);
      }
    }, []);
    if (type === 'ui-options') {
      return (
        <div id={type}>
          {JSON.stringify(props)}
          <span>
            <span />
            <span>
              <span />
            </span>
          </span>
        </div>
      );
    }
    if (type === 'ui-textfield') {
      return (
        <div id={type} onFocus={onFocus as unknown as React.FocusEventHandler}>
          {JSON.stringify(props)}
          <span className="ui-textfield__wrapper__field" />
        </div>
      );
    }
    return <div id={type}>{JSON.stringify(props)}</div>;
  };
}

export const UIP = Component('ui-p');
export const UILink = Component('ui-link');
export const UIImage = Component('ui-image');
export const UITitle = Component('ui-title');
export const UIButton = Component('ui-button');
export const UIOptions = Component('ui-options');
export const UITextarea = Component('ui-textarea');
export const UITextfield = Component('ui-textfield');
export const UIFilePicker = Component('ui-file-picker');
export const generateRandomId = vi.fn(() => '18972182');
export const buildClass = vi.fn((...values: string[]): string => values.join(' '));
export const markdown = vi.fn((label: string, lightMode: boolean) => `MARKDOWN FOR ${label}, ${lightMode}`);
