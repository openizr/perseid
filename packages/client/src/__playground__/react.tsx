/* c8 ignore start */

import {
  Model,
  Store,
  Logger,
  ApiClient,
  FormBuilder,
} from 'scripts/core/index';
import {
  Grid,
  Router,
} from 'scripts/react/index';
import {
  type Ids,
  type Authors,
  type Version,
  type Deletion,
  type Timestamps,
  type DefaultDataModel, I18n,
} from '@perseid/core';
import * as React from 'react';
import '__playground__/style.scss';
import labels from '__playground__/labels';
import { createRoot, type Root } from 'react-dom/client';
import { type StoreSettings } from 'scripts/core/services/Store';
import type { UseSubscription } from '@perseid/store/connectors/react';

let app: Root;

interface DataModel extends DefaultDataModel {
  tests: Ids & Deletion & Timestamps & Authors & Version & {
    elements: null | {
      child: string;
    }[];
  };
}

function main(): void {
  const logger = new Logger({ logLevel: 'debug' });
  const i18n = new I18n(logger, labels);
  const pages: StoreSettings<DataModel>['pages'] = {
    auth: {
      signIn: { route: '/sign-in' },
      signUp: { route: '/sign-up' },
      updateUser: { route: '/users/me' },
      verifyEmail: { route: '/verify-email' },
      resetPassword: { route: '/reset-password' },
    },
    resources: {
      users: {
        list: {
          route: '/users',
          pageProps: {
            searchFields: ['email'],
            fields: [
              '_createdAt',
              'email',
              'roles._createdBy._updatedBy',
              '_devices',
              'password',
              '_apiKeys',
              '_verifiedAt',
            ],
          },
        },
        view: {
          route: '/users/:id',
          pageProps: {
            fields: [
              '_createdAt',
              'roles._createdBy.email',
              '_devices',
              'password',
              '_apiKeys',
              '_verifiedAt',
            ],
          },
        },
        create: {
          route: '/users/create',
          pageProps: {
            fields: ['roles.name'],
          },
        },
        update: {
          route: '/users/:id/edit',
          pageProps: {
            fields: ['roles.name'],
          },
        },
      },
      roles: {
        list: {
          route: '/roles',
          pageProps: {
            fields: ['name'],
            searchFields: ['name'],
          },
        },
        view: {
          route: '/roles/:id',
          pageProps: {
            fields: ['_createdAt', '_createdBy', 'name', 'permissions'],
          },
        },
        update: {
          route: '/roles/:id/edit',
          pageProps: {},
        },
        create: {
          route: '/roles/create',
          pageProps: {},
        },
      },
      tests: {
        list: {
          route: '/tests',
          pageProps: {
            fields: [
              '_createdAt',
              '_createdBy.email',
              'elements',
            ],
          },
        },
        view: {
          route: '/tests/:id',
          pageProps: {
            fields: [
              '_createdAt',
              '_createdBy.email',
              'elements',
            ],
          },
        },
        create: {
          route: '/tests/create',
          pageProps: {
            fields: [],
          },
        },
        update: {
          route: '/tests/:id/edit',
          pageProps: {
            fields: [],
          },
        },
      },
    },
  };
  const model = new Model<DataModel>();
  const apiClient = new ApiClient<DataModel>(model, logger, {
    endpoints: {
      auth: {
        viewMe: { route: '/auth/me' },
        signUp: { route: '/auth/sign-up' },
        signIn: { route: '/auth/sign-in' },
        signOut: { route: '/auth/sign-out' },
        verifyEmail: { route: '/auth/verify-email' },
        refreshToken: { route: '/auth/refresh-token' },
        resetPassword: { route: '/auth/reset-password' },
        requestPasswordReset: { route: '/auth/reset-password' },
        requestEmailVerification: { route: '/auth/verify-email' },
      },
      resources: {
        users: {
          list: { route: '/users' },
          create: { route: '/users' },
          view: { route: '/users/:id' },
          update: { route: '/users/:id' },
          search: { route: '/users/search' },
        },
        roles: {
          list: { route: '/roles' },
          create: { route: '/roles' },
          view: { route: '/roles/:id' },
          update: { route: '/roles/:id' },
          search: { route: '/roles/search' },
        },
        tests: {
          list: { route: '/tests' },
          create: { route: '/tests' },
          view: { route: '/tests/:id' },
          update: { route: '/tests/:id' },
          delete: { route: '/tests/:id' },
          search: { route: '/tests/search' },
        },
      },
    },
    mockedResponses: {},
    connectTimeout: 3000,
    baseUrl: 'http://localhost:5070/perseid',
  });
  const formBuilder = new FormBuilder<DataModel>(model, logger);
  const store = new Store<DataModel>(model, logger, apiClient, formBuilder, {
    pages,
    fallbackPageRoute: '/',
  });
  store.createRoute('/', {
    component: 'Home',
    visibility: 'PRIVATE',
  });
  store.createRoute('/ok', {
    component: 'Home',
    visibility: 'PUBLIC',
  });
  store.createRoute('/ko', {
    component: 'Home',
    visibility: 'PUBLIC_ONLY',
  });
  store.createRoutes();
  const container = document.querySelector('#root') as unknown as HTMLElement;
  app = createRoot(container);
  app.render(
    <React.StrictMode>
      <Router<DataModel>
        services={{
          i18n,
          model,
          store: store as Store<DataModel> & { useSubscription: UseSubscription; },
          apiClient,
        }}
        pages={{
          Home: async () => Promise.resolve({ default: () => <div /> }),
        }}
        components={{}}
      />
      <Grid columns={{ mobile: 4, tablet: 8, desktop: 12 }} />
    </React.StrictMode>,
  );
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting React components tree.
window.addEventListener('beforeunload', () => {
  app.unmount();
});
