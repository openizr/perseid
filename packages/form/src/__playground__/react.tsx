/* c8 ignore start */

import * as React from 'react';
import Form from 'scripts/react/Form';
import { createRoot, Root } from 'react-dom/client';

let app: Root;
const { log } = console;

function main(): void {
  app = createRoot(document.querySelector('#root') as unknown as HTMLElement);
  app.render(
    <React.StrictMode>
      <Form
        configuration={{
          root: 'root',
          fields: {
            test: { type: 'string', required: true },
          },
          steps: {
            root: {
              fields: ['test'],
            },
          },
          plugins: [
            (engine): void => {
              engine.on('userAction', (data, next) => {
                log(data);
                return next(data);
              });
            },
          ],
        }}
        Field={(props): JSX.Element | null => {
          if (props.path === 'root.0.test') {
            return (
              <input onBlur={(event: React.ChangeEvent) => {
                props.engine.userAction({ type: 'input', data: (event.target as HTMLInputElement).value, path: props.path });
              }}
              />
            );
          }
          return null;
        }}
      />
    </React.StrictMode> as unknown as React.ReactNode,
  );
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  app.unmount();
});
