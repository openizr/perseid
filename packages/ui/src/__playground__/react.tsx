/* c8 ignore start */

import * as React from 'react';
import Router from '__playground__/pages/Router';
import { createRoot, Root } from 'react-dom/client';

let app: Root;

function main(): void {
  app = createRoot(document.querySelector('#root') as unknown as HTMLElement);
  app.render(
    <React.StrictMode>
      <Router />
    </React.StrictMode>,
  );
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting React components tree.
window.addEventListener('beforeunload', () => {
  app.unmount();
});
