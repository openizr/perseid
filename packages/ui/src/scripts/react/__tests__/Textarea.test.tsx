/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import React from 'react';
import UITextarea from 'scripts/react/Textarea';
import { render, fireEvent } from '@testing-library/react';

vi.useFakeTimers();
vi.mock('scripts/core/generateRandomId');

describe('react/UITextarea', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const { container } = render(<UITextarea name="test" modifiers="large" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const { container } = render(<UITextarea name="test" id="my-id" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with cols and rows', () => {
    const { container } = render(<UITextarea name="test" cols={10} rows={50} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with autoresize', () => {
    const { container } = render(<UITextarea name="test" autoresize />);
    const textarea = container.getElementsByTagName('textarea')[0];
    fireEvent.change(textarea, { target: { value: 'new\nvalue' } });
    vi.runAllTimers();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with placeholder', () => {
    const { container } = render(<UITextarea name="test" placeholder="test..." />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with label', () => {
    const { container } = render(<UITextarea name="test" label="*Label*" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with helper', () => {
    const { container } = render(<UITextarea name="test" helper="*Helper*" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - with value', () => {
    const { container } = render(<UITextarea name="test" value="my value" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - disabled', () => {
    const onChange = vi.fn();
    const { container } = render(<UITextarea name="test" disabled />);
    const textarea = container.getElementsByTagName('textarea')[0];
    fireEvent.change(textarea, { target: { value: 'new value' } });
    expect(container.firstChild).toMatchSnapshot();
    expect(onChange).not.toHaveBeenCalled();
  });

  test('renders correctly - autocomplete off', () => {
    const { container } = render(<UITextarea name="test" autocomplete="off" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - readonly', () => {
    const onChange = vi.fn();
    const { container } = render(<UITextarea name="test" readonly onChange={onChange} />);
    const textarea = container.getElementsByTagName('textarea')[0];
    fireEvent.change(textarea, { target: { value: 'new value' } });
    expect(container.firstChild).toMatchSnapshot();
    expect(onChange).not.toHaveBeenCalled();
  });

  test('renders correctly - with listeners and debounce', () => {
    const onBlur = vi.fn();
    const onFocus = vi.fn();
    const onChange = vi.fn();
    const onPaste = vi.fn();
    const onKeyDown = vi.fn();
    const { container } = render(<UITextarea
      name="test"
      onBlur={onBlur}
      onFocus={onFocus}
      onPaste={onPaste}
      onChange={onChange}
      onKeyDown={onKeyDown}
      debounceTimeout={250}
    />);
    const textarea = container.getElementsByTagName('textarea')[0];
    fireEvent.focus(textarea);
    fireEvent.blur(textarea);
    fireEvent.keyDown(textarea, { key: 'a' });
    fireEvent.change(textarea, { target: { value: 'new 015 test', selectionStart: 100 } });
    fireEvent.paste(textarea, { clipboardData: { getData: vi.fn(() => 'and 89 OKOK') } });
    vi.runAllTimers();
    expect(container.firstChild).toMatchSnapshot();
    expect(onFocus).toHaveBeenCalledTimes(1);
    expect(onFocus).toHaveBeenCalledWith('', expect.any(Object));
    expect(onBlur).toHaveBeenCalledTimes(1);
    expect(onBlur).toHaveBeenCalledWith('', expect.any(Object));
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('new 015 test', expect.any(Object));
    expect(onKeyDown).toHaveBeenCalledTimes(1);
    expect(onKeyDown).toHaveBeenCalledWith('', expect.any(Object));
    expect(onPaste).toHaveBeenCalledTimes(1);
    expect(onPaste).toHaveBeenCalledWith('new 015 test', expect.any(Object));
  });
});
