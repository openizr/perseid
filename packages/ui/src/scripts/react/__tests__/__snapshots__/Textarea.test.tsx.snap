// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`react/UITextarea > renders correctly - autocomplete off 1`] = `
<div
  class="ui-textarea"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - basic 1`] = `
<div
  class="ui-textarea ui-textarea--large"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - disabled 1`] = `
<div
  class="ui-textarea ui-textarea--disabled"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      disabled=""
      id="z8a8d7a5zad4"
      name="test"
      tabindex="-1"
    />
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - readonly 1`] = `
<div
  class="ui-textarea"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      readonly=""
      tabindex="0"
    />
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - with autoresize 1`] = `
<div
  class="ui-textarea"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      rows="2"
      tabindex="0"
    >
      new
value
    </textarea>
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - with cols and rows 1`] = `
<div
  class="ui-textarea"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      cols="10"
      id="z8a8d7a5zad4"
      name="test"
      rows="50"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - with helper 1`] = `
<div
  class="ui-textarea"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      tabindex="0"
    />
  </div>
  <span
    class="ui-textarea__helper"
  >
    <span
      class="ui-markdown ui-markdown--emphasis"
    >
      Helper
    </span>
  </span>
</div>
`;

exports[`react/UITextarea > renders correctly - with id 1`] = `
<div
  class="ui-textarea"
  id="my-id"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - with label 1`] = `
<div
  class="ui-textarea"
>
  <label
    class="ui-textarea__label"
    for="z8a8d7a5zad4"
  >
    <span
      class="ui-markdown ui-markdown--emphasis"
    >
      Label
    </span>
  </label>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - with listeners and debounce 1`] = `
<div
  class="ui-textarea"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      tabindex="0"
    >
      new 015 test
    </textarea>
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - with placeholder 1`] = `
<div
  class="ui-textarea"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      placeholder="test..."
      tabindex="0"
    />
  </div>
</div>
`;

exports[`react/UITextarea > renders correctly - with value 1`] = `
<div
  class="ui-textarea"
>
  <div
    class="ui-textarea__wrapper"
  >
    <textarea
      autocomplete="off"
      class="ui-textarea__wrapper__field"
      id="z8a8d7a5zad4"
      name="test"
      tabindex="0"
    >
      my value
    </textarea>
  </div>
</div>
`;
